### ISSUE TITLE
Error when downloading file with SFTP fastGet

### ISSUE THREAD

**Issue by morassman** [mscdex/ssh2#327]
I'm getting the following error when downloading a file with either fastGet or createReadStream:
```
events.js:141
      throw er; // Unhandled 'error' event
      ^

Error: stream.push() after EOF
    at readableAddChunk (_stream_readable.js:132:15)
    at SFTPStream.Readable.push (_stream_readable.js:110:10)
    at SFTPStream.Transform.push (_stream_transform.js:128:32)
    at SFTPStream.push (/Users/henkmarais/Temp/iotest/node_modules/ssh2/node_modules/ssh2-streams/lib/sftp.js:177:18)
    at SFTPStream.close (/Users/henkmarais/Temp/iotest/node_modules/ssh2/node_modules/ssh2-streams/lib/sftp.js:811:15)
    at onerror (/Users/henkmarais/Temp/iotest/node_modules/ssh2/node_modules/ssh2-streams/lib/sftp.js:987:13)
    at onread (/Users/henkmarais/Temp/iotest/node_modules/ssh2/node_modules/ssh2-streams/lib/sftp.js:1014:20)
    at state.requests.(anonymous function).cb (/Users/henkmarais/Temp/iotest/node_modules/ssh2/node_modules/ssh2-streams/lib/sftp.js:857:16)
    at SFTPStream._cleanup (/Users/henkmarais/Temp/iotest/node_modules/ssh2/node_modules/ssh2-streams/lib/sftp.js:193:38)
    at SFTPStream.onFinish (/Users/henkmarais/Temp/iotest/node_modules/ssh2/node_modules/ssh2-streams/lib/sftp.js:159:10)
```
Printing out the error gives
```
{ [Error: Malformed packet] level: 'protocol' }
```

I'm using the servers at http://test.rebex.net/ to test with. Some of the files can be downloaded, but some of them cannot. I thought it might be a problem with the server or the files themselves, but everything downloaded properly with another FTP client(muCommander).

The following versions are applicable:
- ssh2 : 0.4.11
- node : 4.1.1
- OSX : 10.10.5

I'm able to reproduce it with the following test program:
```
var config = {};
config.port = 22;
config.host = "test.rebex.net";
config.user = "demo";
config.password = "password";
config.tryKeyboard = true;

SSH2 = require("ssh2");

var client = null;
var ssh2 = new SSH2();

ssh2.on("ready", function() {
  console.log("ssh2.ready");

  ssh2.sftp(function (err, sftp) {
    console.log("sftp");

    sftp.on("end", function() {
      console.log("client.end");
    });

    var path = "/pub/example/KeyGenerator.png";
    var localPath = "./KeyGenerator.png";

    sftp.fastGet(path, localPath, {}, function(err) {
      console.log("fastGet");
      console.error(err);
    });
  });
});

ssh2.on("close", function() {
  console.log("ssh2.close");
});

ssh2.on("error", function(err) {
  console.log("ssh2.error");
  console.error(err);
});

ssh2.on("end", function() {
  console.log("ssh2.end");
});

ssh2.on("keyboard-interactive", function (name, instructions, instructionsLang, prompt, finish) {
  finish(config.password);
});

ssh2.connect(config);
```
Am I perhaps using the library incorrectly?

**Comment by mscdex** [comment]
Ok, so the push after eof is fixed in mscdex/ssh2-streams#1221fe63c and error handling fixed in mscdex/ssh2-streams#5a93bc71a.

However the underlying problem here is that the server is sending a packet larger than the maximum packet size we said we were willing to accept (32768 exactly).

**Comment by morassman** [comment]
Thanks. That was really fast! I'll inform them about the problem.
